app/src/test/java/ch/epfl/qedit/model/MatrixFormatTest.java

```diff
@@ -40,6 +42,19 @@ public void checkMatrix3x3WithParam() {
         checkAllElements(matrixFormat, true, true, 3, 3, 5, "00000");
     }
 
+    @Test
+    public void answerCanBeCorrectlyVisited() {
+        MatrixFormat matrixFormat = MatrixFormat.createMatrix3x3();
+        matrixFormat.accept(
+                new AnswerFormat.Visitor() {
+                    @Override
+                    public void visitMatrixAnswerFormat(MatrixFormat matrixFormat) {
```

Comment:
I think that this test doesn't fail if the matrix format cannot be visited. Maybe you can use Mockito to help you write that test. You first create a mock visitor: `AnswerFormat.Visitor visitor = mock(AnswerFormat.Visitor.class);`. Then, you visit the matrix format: `matrixFormat.accept(visitor);`. Finally, Mockito can verify that the right method was called: `verify(visitor).visitMatrixAnswerFormat(matrixFormat);`.

Reply:
I will not lie to you, i would love if you could write this test.